ports/stm32f3/board_flash.c

```diff
@@ -177,9 +182,40 @@ void board_flash_erase_app(void)
 
 bool board_flash_protect_bootloader(bool protect)
 {
-  // TODO implement later
-  (void) protect;
-  return false;
+  bool ret = true;
+
+  HAL_FLASH_Unlock();
```

Comment:
Will check tomorrow as its been a little while since my head was in the data sheets (will also do the other requested items too).

Reply:
thanks, looking forward to merge this :)

Reply:
https://github.com/STMicroelectronics/stm32f3xx_hal_driver/blob/75cf4fce556ec3b338e3a3798e4b7c1ce20ba83d/Src/stm32f3xx_hal_flash_ex.c#L350-L362

And also looks to be covered on reference manual page 72.

So yes, the flash unlock should be required.

Reply:
ah great, thank you, can you pull modification from  https://github.com/adafruit/tinyuf2/tree/protect and pushed to your branch. Typically, I could push to make and update to pr's fork branch, but somehow I don't have permission to update your fork. Therefore please do update it manually. Then I thik we are good to merge this.

Reply:
I've manually unticked permissions as I view pushing changes to someones PR without previous interaction as a rather hostile way to provide suggestions. 

I'm also not sold on 100% of those changes being tied to the scope of "bootloader protection". However as its your project to manage, I will pull them all if required.

Reply:
please do pull, I often switch between projects and do make some unrelated modification while working on an PR. 
I have no problem without fork without write permission, it is up to invidiuals. I often review rather than pushing, but sometime, a detailed review take more time to write and explain than the actual changes. Also for PR that is pending for too long, and contributors (as you) are probably busy or switched to something else, it may make more sense to push to the pr's branch directly.